client/src/components/logViewer/logParser.ts

```diff
@@ -42,6 +46,8 @@ export function parseLog(logs: LogLine[], logStartFlag: string): Problem[] {
       problemProcessor.appendProblemLogLine(logLine);
       return;
     } else {
+      codeLines.push(logLine.line);
```

Comment:
whether it should be after `if (!isValidSourceCodeLog(logLine))` check below?

Reply:
Empty lines should be added into codeLinesInLog for comparing, because code lines in raw code don't filter those lines.

Reply:
what about those type is not source?

Reply:
only logs with (error | warning | source) type can be into here.

Reply:
ok, cool